Sprint-2/implement/querystring.js

```diff
@@ -1,13 +1,24 @@
 function parseQueryString(queryString) {
   const queryParams = {};
-  if (queryString.length === 0) {
+  if (!queryString) {
     return queryParams;
   }
+
   const keyValuePairs = queryString.split("&");
 
   for (const pair of keyValuePairs) {
-    const [key, value] = pair.split("=");
-    queryParams[key] = value;
+    const index = pair.indexOf("=");
+    let key, value;
+
+    if (index > -1) {
+      key = pair.slice(0, index);
+      value = pair.slice(index + 1);
+    } else {
+      key = pair; // No "=" found; it's a key without a value.
+      value = "";
+    }
+
+    queryParams[key] = decodeURIComponent(value || "");
```

Comment:
Why not just write `... = decodeURIComponent(value)`?

`key` can also be percent or url encoded.